operator/src/main/java/oracle/kubernetes/operator/helpers/BasePodStepContext.java

```diff
@@ -91,6 +93,30 @@ protected V1Container createPrimaryContainer(TuningParameters tuningParameters)
         .securityContext(getServerSpec().getContainerSecurityContext());
   }
 
+  /**
+   * Return a list of environment variables to be set up in the pod. This method does some
+   * processing of the list of environment variables such as token substitution before returning the
+   * list.
+   *
+   * @param tuningParameters TuningParameters containing parameters that may be used in environment
+   *     variables
+   * @return A List of environment variables to be set up in the pod
+   */
+  final List<V1EnvVar> getEnvironmentVariables(TuningParameters tuningParameters) {
+
+    List<V1EnvVar> vars = new ArrayList<>();
+    //List<V1EnvVar> vars = getConfiguredEnvVars(tuningParameters);
+    addConfiguredEnvVarsExcludingAuxImagePaths(tuningParameters, vars);
+
+    addDefaultEnvVarIfMissing(
+            vars, USER_MEM_ARGS, "-Djava.security.egd=file:/dev/./urandom");
+
+    addAuxImagePathsEnvVarIfExists(tuningParameters, vars);
+
+    hideAdminUserCredentials(vars);
+    return doDeepSubstitution(varsToSubVariables(vars), vars);
+  }
+
   protected V1Volume createEmptyDirVolume() {
     V1EmptyDirVolumeSource emptyDirVolumeSource = new V1EmptyDirVolumeSource();
     Optional.ofNullable(getMedium()).ifPresent(emptyDirVolumeSource::medium);
```

Comment:
The BasePodStepContext is shared by `JobStepContext` (used for introspector job pod) and the `PodStepContext` (used for WLS pods) and they both need the emptydir for auxiliary images. This is existing code and not used in conversion webhook.

Reply:
This implies that another level of refactoring would make things more clear in some ways - but perhaps less clear because we already have a bunch of classes.  I'll leave it up to you.

Reply:
ok. The changes to `BasePodStepContext.java` in this PR is only to avoid the pod roll during operator upgrade. When the auxiliary image is created using old approach, AUXILIARY_IMAGE_PATHS env variable is added after all other user specified env variables and some other env variables. When the v8 domain resource is converted to v9, AUXILIARY_IMAGE_PATHS  env variable is now part of the user specified env variable.